app/controllers/users_controller.rb

```diff
@@ -2,6 +2,21 @@ class UsersController < ApplicationController
   def show
```

Comment:
The fetching of the user based on the username is repeated across multiple actions. Consider DRYing (Don't Repeat Yourself) up your code by moving this into a before_action callback that runs before each of these actions